### PR TITLE
Fix Issue 18938 - Dmd segfault when compiling this dub package in test release

### PR DESCRIPTION
**Pull request by RazvanN7**
Foreach on a range that changes its size during iteration strikes again!

### CODE REVIEW

**Comment by RazvanN7** [comment]
buildkite and appveyor still have the old C++ namespace unrelated fail on stable.

File: src/dmd/glue.d [hunk 1]
```diff
@@ -397,8 +397,10 @@ void genObjFile(Module m, bool multiobj)
         m.covb = cast(uint *)calloc((m.numlines + 32) / 32, (*m.covb).sizeof);
     }
 
-    foreach (member; *m.members)
+    const memberSize = m.members.dim;
+    for (int i = 0; i < memberSize; i++)
```
Comment (jacob-carlborg) [inline src/dmd/glue.d]: Can’t a foreach range loop be used?

Comment (ZombineDev) [reply src/dmd/glue.d]: Not 100% sure, but I think all `foreach` AST lowerings cache the range, and we need the bounds to be re-evaluated on every iteration.

Comment (WalterBright) [reply src/dmd/glue.d]: @ZombineDev is right.

Comment (WalterBright) [reply src/dmd/glue.d]: Use `m.members.dim` not `memberSize` for the same reason.

**Review by WalterBright — CHANGES_REQUESTED** [review]

File: src/dmd/glue.d (hunk 1 above)
Comment (RazvanN7) [reply src/dmd/glue.d]: Done

**Review by WalterBright — APPROVED** [review]

**Pull request merged**